cirq-core/cirq/protocols/hash_from_pickle_test.py

```diff
@@ -0,0 +1,107 @@
+    "cirq-core/cirq/protocols/json_test_data/sympy.E.json",
+    "cirq-core/cirq/protocols/json_test_data/sympy.Equality.json",
+    "cirq-core/cirq/protocols/json_test_data/sympy.EulerGamma.json",
+    "cirq-core/cirq/protocols/json_test_data/sympy.Float.json",
+    "cirq-core/cirq/protocols/json_test_data/sympy.GreaterThan.json",
+    "cirq-core/cirq/protocols/json_test_data/sympy.Integer.json",
+    "cirq-core/cirq/protocols/json_test_data/sympy.LessThan.json",
+    "cirq-core/cirq/protocols/json_test_data/sympy.Mul.json",
+    "cirq-core/cirq/protocols/json_test_data/sympy.Pow.json",
+    "cirq-core/cirq/protocols/json_test_data/sympy.Rational.json",
+    "cirq-core/cirq/protocols/json_test_data/sympy.StrictGreaterThan.json",
+    "cirq-core/cirq/protocols/json_test_data/sympy.StrictLessThan.json",
+    "cirq-core/cirq/protocols/json_test_data/sympy.Symbol.json",
+    "cirq-core/cirq/protocols/json_test_data/sympy.Unequality.json",
+    "cirq-core/cirq/protocols/json_test_data/sympy.pi.json",
+    # RigettiQCSAspenDevice does not pickle
+    "cirq-rigetti/cirq_rigetti/json_test_data/RigettiQCSAspenDevice.json",
+    # TODO(#6674,pavoljuhas) - fix pickling of ProjectorSum
+    "cirq-core/cirq/protocols/json_test_data/ProjectorSum.json",
+)
+
+
+def _is_included(json_filename: str) -> bool:
+    json_posix_path = pathlib.PurePath(json_filename).as_posix()
+    if any(json_posix_path.endswith(t) for t in _EXCLUDE_JSON_FILES):
+        return False
+    if not os.path.isfile(json_filename):
+        return False
+    return True
+
+
+@pytest.fixture(scope='module')
+def pool() -> Iterator[multiprocessing.pool.Pool]:
+    ctx = multiprocessing.get_context("spawn")
+    with ctx.Pool(1) as pool:
+        yield pool
+
+
+def _read_json(json_filename: str) -> Any:
+    obj = cirq.read_json(json_filename)
+    obj = obj[0] if isinstance(obj, list) else obj
+    # trigger possible caching of the hash value
+    if isinstance(obj, Hashable):
+        _ = hash(obj)
+    return obj
+
+
+def test_exclude_json_files_has_valid_entries() -> None:
+    """Verify _EXCLUDE_JSON_FILES has valid entries."""
+    invalid_json_paths = [f for f in _EXCLUDE_JSON_FILES if not REPO_ROOT.joinpath(f).is_file()]
+    assert invalid_json_paths == []
+
+
+@pytest.mark.parametrize(
+    'json_filename',
+    [
+        f"{abs_path}.json"
+        for m in MODULE_TEST_SPECS
+        for abs_path in m.all_test_data_keys()
+        if _is_included(f"{abs_path}.json")
+    ],
+)
+def test_hash_from_pickle(json_filename: str, pool: multiprocessing.pool.Pool):
+    obj_local = _read_json(json_filename)
+    if not isinstance(obj_local, Hashable):
+        return
+    # check if pickling works in single process for the sake of debugging
+    obj_copy = pickle.loads(pickle.dumps(obj_local))
+    assert obj_copy == obj_local
+    assert hash(obj_copy) == hash(obj_local)
+    obj_worker = pool.apply(_read_json, [json_filename])
```

Comment:
A comment here might help to explain what's going on
```suggestion

    # Read and hash the object in a separate process, then send it back
    # which requires pickling and unpickling.
    obj_worker = pool.apply(_read_json, [json_filename])
```